index.js

```diff
@@ -12,7 +12,58 @@ function setMultipart (req, done) {
   done()
 }
 
+function attachToBody (options, req, reply, next) {
+  req[kMultipart] = true
+
+  const defaultConsumer = (field, file, filename, encoding, mimetype) => {
+    const fileData = []
+    file.on('data', data => { fileData.push(data) })
+    file.on('limit', () => { body[field].limit = true })
+    file.on('end', () => {
+      body[field].data = Buffer.concat(fileData)
+    })
+  }
+
+  const consumerStream = options.onFile || defaultConsumer
+  const body = { }
+  const mp = req.multipart((field, file, filename, encoding, mimetype) => {
+    body[field] = {
+      data: [],
+      filename,
+      encoding,
+      mimetype,
+      limit: false
+    }
+    consumerStream(field, file, filename, encoding, mimetype)
```

Comment:
You might want to check if this returns a promise and attach a .catch() handler.